File: backend/aci/common/schemas/project.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field

from aci.common.enums import Visibility
from aci.common.schemas.agent import AgentPublic


class ProjectCreate(BaseModel):
    """Project can be created under a user or an organization."""

    name: str
    org_id: UUID = Field(
        description="Organization ID if project is to be created under an organization",
    )


class ProjectPublic(BaseModel):
    id: UUID
    org_id: UUID
    name: str
    visibility_access: Visibility
    daily_quota_used: int
    daily_quota_reset_at: datetime
    total_quota_used: int

    created_at: datetime
    updated_at: datetime

    agents: list[AgentPublic]

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/lib/types/project.ts
export interface Project {
  id: string;
  owner_id: string;
  name: string;
  visibility_access: string;
  daily_quota_used: number;
  daily_quota_reset_at: string;
  total_quota_used: number;
  created_at: string;
  updated_at: string;
  agents: Agent[];
}

export interface Agent {
  id: string;
  project_id: string;
  name: string;
  description: string;
  allowed_apps: string[];
  custom_instructions: Record<string, string>;
  created_at: string;
  updated_at: string;
  api_keys: APIKey[];
}

export interface APIKey {
  id: string;
  key: string;
  agent_id: string;
  status: string;
  created_at: string;
  updated_at: string;
}
